tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordSession.cs

```diff
@@ -167,7 +198,21 @@ public async Task Sanitize(IEnumerable<RecordedTestSanitizer> sanitizers, bool s
             {
                 foreach (var sanitizer in sanitizers)
                 {
+                    var entriesPreSanitize = Array.Empty<RecordEntry>();
+                    if (DebugLogger.CheckLogLevel(LogLevel.Debug))
+                    {
+                        entriesPreSanitize = this.Entries.Select(requestEntry => { requestEntry.initializeIsModifiedFlag(); return CopyRecordEntry(requestEntry); }).ToArray();
+                    }
+                    
                     sanitizer.Sanitize(this);
+
+                    if (DebugLogger.CheckLogLevel(LogLevel.Debug))
+                    {
+                        for (int i = 0; i < entriesPreSanitize.Length; i++)
```

Comment:
Check for entriesPostSanitize[i] actually existing. Remember that sanitizers can clean out an entire entry.

Just looking to avoid out of index exception.

Reply:
Will do.
That's interesting, I didn't realize `this.Entries[i]` can be cleaned out entirely, can you give me an example/workflow where/how that happens.

Reply:
There are couple ways you could do this.

```
for (int i = 0; i < entriesPreSanitize.Length; i++)
```

could be become

```
var max = Math.Max(entriesPreSanitize.Length, entries.Length)
for (int i = 0; i <max ; i++)
```

and then check that the two arrays have equal length. If they aren't that is also a modification 👍

Reply:
Is it possible that both the arrays go out of sequence too?
Because the logging diff is for each entry of the array and not just the whole array at once.

Reply:
handled all the <, >, = cases individually.
- throws if the entries increase
- does not log entries/modifications if entries decrease
- if entry count remains the same, logs the diff